src/builder/guard.rs

```diff
@@ -100,6 +104,31 @@ impl<'a> Guard<'a> {
                 "Error creating dir: {err}");
         }
 
+        if !self.ctx.container_config.remove_all_except.is_empty() {
+            let root = Path::new("/vagga/root");
+            let exclude_paths = PROTECTED_DIRS.iter()
```

Comment:
> Why exclude_paths is a Vec<PathBuf> if it could be Vec<&Path>? Or just Iterator<&Path> (actually a Chain.. but that's irrelevant)

With iterator there was a lifetime issue that I couldn't resolve.